src/metabase/task/sync_databases.clj

```diff
@@ -29,20 +29,23 @@
   [job-context]
   (Database (u/get-id (get (qc/from-job-data job-context) "db-id"))))
 
-
-(jobs/defjob SyncAndAnalyzeDatabase [job-context]
-  (let [database (job-context->database job-context)]
-    (sync-metadata/sync-db-metadata! database)
-    ;; only run analysis if this is a "full sync" database
-    (when (:is_full_sync database)
-      (analyze/analyze-db! database))))
-
-
-(jobs/defjob UpdateFieldValues [job-context]
-  (let [database (job-context->database job-context)]
-    (when (:is_full_sync database)
-      (field-values/update-field-values! (job-context->database job-context)))))
-
+;; The DisallowConcurrentExecution on the two defrecords below attaches an annotation to the generated class that will
+;; constrain the job execution to only be one at a time. Other triggers wanting the job to run will misfire.
+(defrecord ^{DisallowConcurrentExecution true} SyncAndAnalyzeDatabase []
```

Comment:
If you just make this `^{org.quartz.DisallowConcurrentExecution true}` won't it still work even if we accidentally remove the `:import`?

Reply:
Did adding the `DisallowConcurrentExecution` metadata not work when using the `defjob` macro?

Reply:
I didn't think the metadata would convey through the `defjob` but I just tested it out locally and it looks like it works. I'll switch it over. I'll fully qualify the annotation as well.

Reply:
yeah I don't 100% understand the implementation details but metadata carries through on symbols when used in macros, which is why for example you can do something like make a multimethod `^:private` (`defmulti` itself is a macro). I suppose it has something to do with the way the metadata is read in because if you want to set metadata in a macro itself you have do something to like use `vary-meta`, e.g.

```clojure
;; this works
(defmacro def- [symb & more]
  `(def ~(vary-meta symb assoc :private true)
     ~@more))

(def- one-hundred 100)
(:private (meta #'one-hundred)) ; -> true

;; this doesn't work 
(defmacro def-2 [symb & more]
  `(def ^:private ~symb ~@more))

(def-2 two-hundred 200)
(:private (meta #'two-hundred)) ; -> nil

;; but metadata applied to the symbol outside of the macro will still carry over
(def-2 ^:private three-hundred 300)
(:private (meta #'three-hundred)) ; -> true
```